docs/rules/jsx-newline.md

```diff
@@ -0,0 +1,65 @@
+# Enforce a new line after jsx elements and expressions (react/jsx-newline)
+
+**Fixable:** This rule is automatically fixable using the `--fix` flag on the command line.
+
+## Rule Details
+
+This is a stylistic rule intended to make JSX code more readable by enforcing spaces between adjacent JSX elements and expressions.
+
+The following patterns are considered warnings:
+
+```jsx
+<div>
+  <Button>{data.label}</Button>
+  <List />
+</div>
+```
+
+```jsx
+<div>
+  <Button>{data.label}</Button>
+  {showSomething === true && <Something />}
+</div>
+```
+
+```jsx
+<div>
+  {showSomething === true && <Something />}
+  {showSomethingElse === true ? (
+    <SomethingElse />
+  ) : (
+    <ErrorMessage />
+  )}
+</div>
+```
+
+The following patterns are **not** considered warnings:
+
+```jsx
+<div>
+  <Button>{data.label}</Button>
+
+  <List />
+
+  <Button>
+    <IconPreview />
+    Button 2
```

Comment:
how is this not a warning? there's no newline after the element.

Reply:
In the **Good** part of the issue description the same example is giving without a newline. That is why the rule only warns about newlines between elements and expressions, no literals like text and numbers.